### PR TITLE
Quick prototype for not being able to

### PR DESCRIPTION
**Pull request by coverbeck**
create services and workflows with the same path

Just tried one simple test so far, manually. 

@denis-yuen if you have a chance to glance at before I start my day, let me know what you think of this approach -- sort of what we discussed, although maybe not as clean as I would like.

### CODE REVIEW

File: dockstore-webservice/src/main/java/io/dockstore/webservice/jdbi/WorkflowDAO.java [hunk 1]
```diff
@@ -134,7 +137,24 @@ public Workflow findByPath(String path, boolean findPublished) {
...
+     *
+     * @param path
+     * @param findPublished
+     * @param clazz
+     * @param <T>
+     * @return
+     */
+    public <T extends Workflow> Optional<T> findByPath(String path, boolean findPublished, Class<T> clazz) {
+        final List<Workflow> workflows = findByPath(path, findPublished);
+        return workflows.stream().filter(w -> w.getClass().equals(clazz)).map(clazz::cast).findFirst();
```
Comment (denis-yuen) [inline dockstore-webservice/src/main/java/io/dockstore/webservice/jdbi/WorkflowDAO.java]: Maybe this should throw an exception if the stream has more than one after the filter()

File: dockstore-webservice/src/main/java/io/dockstore/webservice/resources/WorkflowResource.java [hunk 2]
```diff
@@ -553,7 +551,7 @@ public Workflow upsertServiceVersion(@ApiParam(hidden = true) @Auth User user,
         String dockstoreServicePath = upsertVersionHelper(repository, gitReference, null, WorkflowMode.SERVICE, installationAccessToken);
 
         // Add user to service if necessary
-        Workflow service = workflowDAO.findByPath(dockstoreServicePath, false);
+        Workflow service = workflowDAO.findByPath(dockstoreServicePath, false, Service.class).get();
```
Comment (denis-yuen) [inline dockstore-webservice/src/main/java/io/dockstore/webservice/resources/WorkflowResource.java]: Maybe this should be in a new class (with the same endpoint path for UI compatibility) for services

Comment (coverbeck) [reply dockstore-webservice/src/main/java/io/dockstore/webservice/resources/WorkflowResource.java]: Do you mean I should have `serviceDAO.findByPath`?

And/or do you mean I should have `ServiceResource.upsertServiceVersion`?

Comment (denis-yuen) [reply dockstore-webservice/src/main/java/io/dockstore/webservice/resources/WorkflowResource.java]: The latter

**Comment by coverbeck** [comment]
Will reopen as a real PR

**Pull request closed without merging**